src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -118,7 +118,7 @@ end
 # Perform a Rayleigh-Ritz for the N first eigenvectors.
 @timing function rayleigh_ritz(X, AX, N)
     XAX = X' * AX
-    @assert all(!isnan, XAX)
+    @assert !any(isnan, XAX)
```

Comment:
any particular reason for this?

Reply:
Yes. On cuda funnily the two statements are not equal. I had the original assertion fail even without nans in the array.

Reply:
What? Sounds like a bug that should be reported?

Reply:
Yeah, probably. In any case we very easily get NaNs when using CUDA, so we need to give this a closer look anyway.